### PR TITLE
bugfix/COR-1240-Timeseries-chart-label-goes-off-screen-on-mobile

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard28**
# Summary

This PR fixes the issue where x-axis end labels were falling outside of the viewport on mobile AND falling out of grid on desktop. The issue was caused due to the end label being `centered` when it is a short label (without year). Because of this, the label exceeded the grid. This has been changed so that end labels are always aligned to the `end` which means to the left of the `tick`.

### BEFORE
BEFORE

### AFTER
AFTER

### CODE REVIEW

File: packages/app/src/components/time-series-chart/components/axes.tsx [hunk 1]
```diff
@@ -119,143 +108,70 @@ export const Axes = memo(function Axes<T extends TimestampedValue>({
...
-        ? hasDatesAsRange
-          ? 6
-          : 4
-        : hasDatesAsRange
-        ? 5
-        : 3
-      : hasDatesAsRange
-      ? 3
-      : 2;
+    const preferredDateTicks = breakpoints.sm ? (timeframe === 'all' ? (hasDatesAsRange ? 6 : 4) : hasDatesAsRange ? 5 : 3) : hasDatesAsRange ? 3 : 2;
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard28) [reply packages/app/src/components/time-series-chart/components/axes.tsx:124]: Yeah Its not very clear at all, however I did not touch this piece of code, its formatting. If i try to put it back, then the formatter will put it on one line again. If you want, I can try to simplify it.

[the remark this answers is not in the document]
Comment (VWSCoronaDashboard19) [reply packages/app/src/components/time-series-chart/components/axes.tsx:124]: Yes please

[the remark this answers is not in the document]
Comment (VWSCoronaDashboard25) [reply packages/app/src/components/time-series-chart/components/axes.tsx:124]: In regards of the axes component, I like that the details of how many date ticks are preferred are put away. I'd argue this makes it easier to understanding the axes component, because those details are irrelevant.

In regards of the amount of preferred date ticks, I agree the logic is (still) hard to follow, but it's hopefully rare that we as developers need to dive into this logic, but it's doable. So I'd argue to keep it on one line.

I would like the variable to be typed though, so that it's more easy to see it's just stuff to get a `number`.

```suggestion
    const preferredDateTicks: number = breakpoints.sm ? (timeframe === 'all' ? (hasDatesAsRange ? 6 : 4) : hasDatesAsRange ? 5 : 3) : hasDatesAsRange ? 3 : 2;
```

File: packages/app/src/components/time-series-chart/components/axes.tsx [hunk 2]
```diff
@@ -283,19 +198,10 @@ export const Axes = memo(function Axes<T extends TimestampedValue>({
 
   /**
    * Using anchor middle the line marker label will fall nicely on top
-   * of the axis label.
-   *
-   * The only times at which we can not use middle is if we are
-   * rendering a year in the label, because it becomes too long.
+   * of the axis label. This will only happen for labels which are not the first or last label.
```
Comment (VWSCoronaDashboard25) [inline packages/app/src/components/time-series-chart/components/axes.tsx:201]: No need to split the comment mid-sentence anymore.

```typescript
/**
 * Using anchor middle the line marker label will fall nicely on top of the axis label.
 * This will only happen for labels which are not the first or last label.
 */
```

File: packages/app/src/components/time-series-chart/components/axes.tsx [hunk 3]
```diff
@@ -265,7 +181,6 @@ export const Axes = memo(function Axes<T extends TimestampedValue>({
    * a space plus a three character month, which makes 6.
    */
   const isLongStartLabel = xTicks[0].length > 6;
```
Comment (VWSCoronaDashboard25) [inline packages/app/src/components/time-series-chart/components/axes.tsx:183]: This variable is not needed in the next section, so I think it's best to move it down just above the first section where it's used.

Comment (VWSCoronaDashboard25) [reply packages/app/src/components/time-series-chart/components/axes.tsx:183]: It's even only used anywhere else I think, so best to stuff it inside the function, I guess.

```typescript
const getAnchor = (x: NumberValue) => {
  const isLongStartLabel = xTicks[0].length > 6;
  return x === tickValues[0] && isLongStartLabel ? (tickValues.length === 1 ? 'middle' : 'start') : x === tickValues[tickValues.length - 1] ? 'end' : 'middle';
```

**Review by VWSCoronaDashboard19 — APPROVED** [review]

File: packages/app/src/components/time-series-chart/components/axes.tsx [hunk 4]
```diff
@@ -201,7 +194,19 @@ export const Axes = memo(function Axes<T extends TimestampedValue>({
...
+
+    if (isFirstTick && isLongStartLabel && tickValues.length !== 1) {
+      return 'start';
+    }
+
+    if (isLastTick) {
+      return 'end';
+    }
+
+    return 'middle';
```
Comment (VWSCoronaDashboard25) [inline packages/app/src/components/time-series-chart/components/axes.tsx:209]: Thanks! This is much better!

**Review by VWSCoronaDashboard25 — APPROVED** [review]
Well done!
